database/src/test/java/database/service/SqlDatabaseServiceTest.java

```diff
@@ -0,0 +1,274 @@
+                sqlDatabaseService.createTable(entityClass));
+        assertEquals(UNABLE_CREATE_TABLE, exception.getMessage());
+    }
+
+    @Test
+    void deleteTableTest() throws SQLException {
+        Class<? extends BaseEntity> entityClass = Student.class;
+        sqlDatabaseService.createTable(entityClass);
+
+        assertTrue(sqlDatabaseService.deleteTable(entityClass));
+        assertFalse(sqlDatabaseService.checkTableExists(DATABASE_NAME, "Student"));
+    }
+
+    @Test
+    void addNewRecordToTablePositiveTest() throws SQLException {
+        Class<? extends BaseEntity> entityClass = Student.class;
+
+        sqlDatabaseService.createTable(entityClass);
+
+        Student receivedFirstStudent = sqlDatabaseService.addNewRecordToTable(firstStudent);
+        assertEquals(firstStudent, receivedFirstStudent);
+        assertEquals(1, receivedFirstStudent.getId());
+
+        Student receivedSecondStudent = sqlDatabaseService.addNewRecordToTable(secondStudent);
+        assertEquals(secondStudent, receivedSecondStudent);
+        assertEquals(2, receivedSecondStudent.getId());
+    }
+
+    @Test
+    void addNewRecordWhenDatabaseDoesNotExistTest() {
+        DatabaseDoesNotExistException exception = assertThrows(DatabaseDoesNotExistException.class, () ->
+                sqlDatabaseService.addNewRecordToTable(firstStudent));
+
+        assertEquals(TABLE_NOT_EXIST, exception.getMessage());
+    }
+
+    @Test
+    void addNewRecordIdProvidedManually() {
+        sqlDatabaseService.createTable(Student.class);
+
+        Student studentWithManuallyProvidedId = new Student("FirstName1 LastName1", 5.0);
+        studentWithManuallyProvidedId.setId(7);
+
+        IdProvidedManuallyException exception = assertThrows(IdProvidedManuallyException.class, () ->
+                sqlDatabaseService.addNewRecordToTable(studentWithManuallyProvidedId));
+
+        assertEquals(ID_PROVIDED_MANUALLY, exception.getMessage());
+    }
+
+    @Test
+    void updateRecordInTableWIthCorrectIdTest() {
+        sqlDatabaseService.createTable(Student.class);
+        sqlDatabaseService.addNewRecordToTable(firstStudent);
+        sqlDatabaseService.addNewRecordToTable(thirdStudent);
+        //Check that original Id is not changed
+        secondStudent.setId(5);
+
+        assertEquals(secondStudent, sqlDatabaseService.updateRecordInTable(secondStudent, 1));
+        assertEquals(secondStudent, sqlDatabaseService.getById(Student.class, 1));
+    }
+
+    @Test
+    void updateOxfordStudentTest() {
+        sqlDatabaseService.createTable(OxfordStudent.class);
+        OxfordStudent os = new OxfordStudent("N", 4.5, 20);
+        OxfordStudent os2 = new OxfordStudent("M", 4.2, 21);
+
+        sqlDatabaseService.addNewRecordToTable(os);
+        assertEquals(os2, sqlDatabaseService.updateRecordInTable(os2, 1));
+
+        sqlDatabaseService.deleteTable(OxfordStudent.class);
```

Comment:
Please use `try-finally` block loke you do it in `removeAllRecordsResetIdCounterCheckTest` method.